src/fu-engine.c

```diff
@@ -3937,28 +4049,28 @@ fu_engine_add_approved_firmware (FuEngine *self, const gchar *checksum)
 gchar *
 fu_engine_self_sign (FuEngine *self,
 		     const gchar *value,
-		     FuKeyringSignFlags flags,
+		     JcatSignFlags flags,
 		     GError **error)
 {
-	g_autoptr(FuKeyring) kr = NULL;
-	g_autoptr(FuKeyringResult) kr_result = NULL;
+	g_autoptr(JcatEngine) jcat_engine = NULL;
+	g_autoptr(JcatResult) jcat_result = NULL;
 	g_autoptr(GBytes) payload = NULL;
 	g_autoptr(GBytes) signature = NULL;
 
 	/* create detached signature and verify */
-	kr = fu_keyring_create_for_kind (FWUPD_KEYRING_KIND_PKCS7, error);
-	if (kr == NULL)
-		return NULL;
-	if (!fu_keyring_setup (kr, error))
+	jcat_engine = jcat_context_get_engine (self->jcat_context,
+					       FWUPD_KEYRING_KIND_PKCS7,
```

Comment:
Isn't this the wrong enum type?  It should be a jcat type.

Reply:
Great catch! I deliberately gave them the same values for GPG and PKCS, but this is still a valid bug.